From 3953f56f98420e2ecb0e1c4542e20bfbb81da965 Mon Sep 17 00:00:00 2001
From: Takuto Ikuta <tikuta@google.com>
Date: Mon, 9 Dec 2024 18:45:36 +0900
Subject: [PATCH] add missing includes for the build with use_libcxx_modules

This is to fix build error when we set use_libcxx_modules=true in
chromium build.

Bug: 40440396
Change-Id: I0cca5190c2c3551cc62bafafd12da9283e4afdd4

Upstream-Status: Backport [3953f56 add missing includes for the build with use_libcxx_modules]
- Removed some patches for files that do not exist in the current version v47.0

Signed-off-by: mark.yang <mark.yang@lge.com>
---

diff --git a/include/perfetto/base/flat_set.h b/include/perfetto/base/flat_set.h
index 9390537b39..5be590ea6b 100644
--- a/include/perfetto/base/flat_set.h
+++ b/include/perfetto/base/flat_set.h
@@ -17,7 +17,10 @@
 #ifndef INCLUDE_PERFETTO_BASE_FLAT_SET_H_
 #define INCLUDE_PERFETTO_BASE_FLAT_SET_H_
 
+#include <stddef.h>
+
 #include <algorithm>
+#include <utility>
 #include <vector>
 
 // A vector-based set::set-like container.
diff --git a/include/perfetto/base/time.h b/include/perfetto/base/time.h
index 6436503d20..8de7648267 100644
--- a/include/perfetto/base/time.h
+++ b/include/perfetto/base/time.h
@@ -17,6 +17,7 @@
 #ifndef INCLUDE_PERFETTO_BASE_TIME_H_
 #define INCLUDE_PERFETTO_BASE_TIME_H_
 
+#include <stdint.h>
 #include <time.h>
 
 #include <chrono>
diff --git a/include/perfetto/ext/base/string_utils.h b/include/perfetto/ext/base/string_utils.h
index c8264a2fb7..3b15e55771 100644
--- a/include/perfetto/ext/base/string_utils.h
+++ b/include/perfetto/ext/base/string_utils.h
@@ -24,6 +24,7 @@
 #include <cinttypes>
 #include <optional>
 #include <string>
+#include <system_error>
 #include <vector>
 
 #include "perfetto/ext/base/string_view.h"
diff --git a/include/perfetto/ext/base/subprocess.h b/include/perfetto/ext/base/subprocess.h
index 12121e0284..4075169df1 100644
--- a/include/perfetto/ext/base/subprocess.h
+++ b/include/perfetto/ext/base/subprocess.h
@@ -20,6 +20,7 @@
 #include <condition_variable>
 #include <functional>
 #include <initializer_list>
+#include <memory>
 #include <mutex>
 #include <optional>
 #include <string>
diff --git a/include/perfetto/ext/base/waitable_event.h b/include/perfetto/ext/base/waitable_event.h
index 0e786196f8..a4b268a2f3 100644
--- a/include/perfetto/ext/base/waitable_event.h
+++ b/include/perfetto/ext/base/waitable_event.h
@@ -17,6 +17,8 @@
 #ifndef INCLUDE_PERFETTO_EXT_BASE_WAITABLE_EVENT_H_
 #define INCLUDE_PERFETTO_EXT_BASE_WAITABLE_EVENT_H_
 
+#include <stdint.h>
+
 #include <condition_variable>
 #include <mutex>
 
diff --git a/include/perfetto/ext/tracing/core/slice.h b/include/perfetto/ext/tracing/core/slice.h
index 063043bdc6..cdabc39465 100644
--- a/include/perfetto/ext/tracing/core/slice.h
+++ b/include/perfetto/ext/tracing/core/slice.h
@@ -18,6 +18,7 @@
 #define INCLUDE_PERFETTO_EXT_TRACING_CORE_SLICE_H_
 
 #include <stddef.h>
+#include <stdint.h>
 #include <string.h>
 
 #include <memory>
diff --git a/src/trace_processor/util/trace_blob_view_reader.h b/src/trace_processor/util/trace_blob_view_reader.h
index c39ffaba7c..6ffa7ff050 100644
--- a/src/trace_processor/util/trace_blob_view_reader.h
+++ b/src/trace_processor/util/trace_blob_view_reader.h
@@ -19,6 +19,7 @@
 
 #include <cstddef>
 #include <optional>
+#include <vector>
 
 #include "perfetto/ext/base/circular_queue.h"
 #include "perfetto/trace_processor/trace_blob_view.h"
